platform/core.startup/src/org/netbeans/core/startup/ModuleSystem.java

```diff
@@ -176,54 +176,69 @@ public void loadBootModules() {
         ev.log(Events.START_LOAD_BOOT_MODULES);
         try {
             bootModules = new HashSet<Module>(10);
-            ClassLoader loader = ModuleSystem.class.getClassLoader();
+            ClassLoader upperLoader = ModuleSystem.class.getClassLoader();
+            // wrap alien loader, so it can be used among parent loaders of module (instanceof ProxyClassLoader)
+            ClassLoader loader = new JarClassLoader(Collections.<File>emptyList(), new ClassLoader[] { Lookup.class.getClassLoader() });
```

Comment:
Maybe we could stick to module system classes and use `Module.class.getClassLoader()` - the result should be the same as both `boot.jar` and `org-openide-util-lookup.jar` are in the `platform/lib` - however the `Module` and `ModuleSystem` are more "co-located" than `Lookup`.

Just a stylistic change, however.